Simplify head behaviour (#828)
NUbots/NUbots: module/motion/HeadController/README.md

## Usage

Takes a HeadCommand message:

- If command requests smoothing (`command->smooth = true`), then the goal angle emitted to is smoothed using exponential filter, tunable using `smoothing_factor`
- If command request is in world space (`command->goalRobotSpace = false`) the goal angle is converted from world space to robot space

The final goal angle is clamped based on the maximum/minium pitch and yaw of the head servos before being sent to servo controller

## Consumes

* `message::motion::HeadCommand` command containing desired angles, smoothing flag and reference frame information

> 

Reviewer: Also consumes KinematicsModel

## Emits

> *`message::behaviour::ServoCommands` instructs servo controller to move head to desired angle

Reviewer: ```suggestion
* `message::behaviour::ServoCommands` instructs servo controller to move head to desired angle
```

> 

Reviewer: it emits a headcommand too (at the beginning), probably explain this.
emits a nusight graph, probably mention this
it also emits a registeraction, mention this (it registers the module to be allowed to move the servos, and it specifically only registers for the head and gives itself a high priority of 30 (more is higher priority). if it loses priority of the head, it disables the main reaction loop and if it gets back priority it enables the reaction again).

*`message::behaviour::ServoCommands` instructs servo controller to move head to desired angle

## Dependencies

> 

Reviewer: Eigen, IK, some maths stuff too

## Usage

Takes a HeadCommand message:

- If command requests smoothing (`command->smooth = true`), then the goal angle emitted to is smoothed using exponential filter, tunable using `smoothing_factor`
- If command request is in world space (`command->goalRobotSpace = false`) the goal angle is converted from world space to robot space

The final goal angle is clamped based on the maximum/minium pitch and yaw of the head servos before being sent to servo controller

## Consumes

* `message::motion::HeadCommand` command containing desired angles, smoothing flag and reference frame information
* `message::input::Sensors` contains information to map desired angles into robot frame from world frame
* `message::motion::KinematicsModel` contains maximum and minimum values for head pitch/yaw

## Emits

* `message::motion::HeadCommand` instructs servo controller to move head to initial position when configuring
* `message::behaviour::ServoCommands` instructs servo controller to move head to desired goal angle

> * `utility::behaviour::RegisterAction` it registers the module to be allowed to move the servos, and it specifically only registers for the head and gives itself a high priority of 30 (more is higher priority). if it loses priority of the head, it disables the main reaction loop and if it gets back priority it enables the reaction again).

Reviewer: ```suggestion
* `utility::behaviour::RegisterAction` registers the module so it is allowed to move the servos. It registers for access to the head with a priority of 30 (a higher number is a higher priority). If it loses priority of the head, it disables the main reaction loop and if it gets back priority it enables the reaction again.
```
I'm still not sure if I like my wording of this but this is probably a bit nicer.
